arch/arm/src/armv7-m/arm_schedulesigaction.c

```diff
@@ -155,7 +146,20 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
                * as the interrupt return context.
                */
 
-              arm_savestate(tcb->xcp.regs);
+              arm_savestate(tcb->xcp.saved_regs);
+
```

Comment:
Move the new block before line 134 since we must save origin context before setup to sigdeliver

Reply:
Yes, which is already resolved on my local